code/game/machinery/doors/door.dm

```diff
@@ -272,6 +315,45 @@
 		emagged = TRUE
 		return TRUE
 
+/obj/machinery/door/cmag_act(mob/user)
+	if(!density)
+		return
+	flick("door_spark", src)
+	sleep(6) //The cmag doesn't automatically open doors. It inverts access, not provides it!
+	cmagged = TRUE
+	return TRUE
+
+//Proc for inverting access on cmagged doors."canopen" should always return the OPPOSITE of the normal result.
+/obj/machinery/door/proc/cmag_switch(canopen, mob/living/user)
```

Comment:
Can you make this proc use early returns?

Reply:
Done.